models.py

```diff
@@ -6,9 +6,45 @@
 
 from data_pipeline.glove_embeddings import get_weight_matrix
 
+# this size came from the highest code of a character that was found in the data set
+# の has the code 12398
+NUMBER_OF_ASCII_CHARACTERS = 12400
+
 #TODO: move this.
 BOS_IDX = 1
 
+class CharacterLevelEmbedding(nn.Module):
+
+
+  def __init__(self, config: CfgNode):
+    super(CharacterLevelEmbedding, self).__init__()
+    self.embedding = nn.Embedding(NUMBER_OF_ASCII_CHARACTERS, config.CHAR_EMB_DIM)
+    self.gru = nn.GRU(config.CHAR_EMB_DIM, config.CHAR_HIDDEN_SIZE)
+    self.char_hidden_size = config.CHAR_HIDDEN_SIZE
+
+  def forward(self, inputs: torch.Tensor, input_lengths: torch.Tensor):
+    """
+    Compute the learned embeddings at character level
+    :param inputs: (torch.Tensor): Inputs (word len, input seq len, batch size).
+           input_lengths: (torch.Tensor): Inputs Length (input seq len, batch size)
+    :return:
+    """
+    # word_len x input_seq_len x batch_size x char_emb_dim
+    embedded_inputs = self.embedding(inputs)
+    word_len, input_seq_len, batch_size, char_emb_size = embedded_inputs.shape
+    packable_embedded_inputs = embedded_inputs.view(word_len, input_seq_len * batch_size, char_emb_size)
+    packable_input_lengths = input_lengths.reshape(input_seq_len * batch_size)
```

Comment:
why not use view? do you need reshape?

Reply:
If am using view I get an exception: RuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.
After changing it to reshape, it worked.

Reply:
yes, but do you understand what that means? Are you just hiding a bug? In my mind the shape should be good, we should not need reshape. Have you checked that the number of elements is good? Reshape is more expensive and if all works well I don't see why we should need it